internal/ingress/controller/parser/parser_test.go

```diff
@@ -4004,6 +4004,61 @@ func TestValidateProtocol(t *testing.T) {
 		overrideUpstream(nil, nil, make(map[string]string))
 	})
 }
+func TestUseSSLProtocol(t *testing.T) {
+	assert := assert.New(t)
+	testTable := []struct {
+		inRoute  kong.Route
+		outRoute kong.Route
+	}{
+		{
+			kong.Route{
+				Protocols: kong.StringSlice("grpc", "grpcs"),
+			},
+			kong.Route{
+				Protocols: kong.StringSlice("grpcs"),
+			},
+		},
+		{
+			kong.Route{
+				Protocols: kong.StringSlice("http", "https"),
+			},
+			kong.Route{
+				Protocols: kong.StringSlice("https"),
+			},
+		},
+		{
+			kong.Route{
+				Protocols: kong.StringSlice("grpcs", "https"),
+			},
+
+			kong.Route{
+				Protocols: kong.StringSlice("grpcs", "https"),
+			},
+		},
+		{
+			kong.Route{
+				Protocols: kong.StringSlice("grpc", "http"),
+			},
+			kong.Route{
+				Protocols: kong.StringSlice("grpcs", "https"),
+			},
```

Comment:
nit: A mix of grpc and http is not possible. Have this test case here is fine but useless, meaning it adds unnecessary overheads.

Reply:
Will clean it up and raise another PR.